lib/fixup.rb

```diff
@@ -12,3 +12,104 @@
+  { pkg_name: 'qtwebengine', pkg_rename: 'qt5_webengine', pkg_deprecated: nil },
+  { pkg_name: 'qtwebglplugin', pkg_rename: 'qt5_webglplugin', pkg_deprecated: nil },
+  { pkg_name: 'qtwebsockets', pkg_rename: 'qt5_websockets', pkg_deprecated: nil },
+  { pkg_name: 'qtx11extras', pkg_rename: 'qt5_x11extras', pkg_deprecated: nil },
+  { pkg_name: 'qtchooser', pkg_rename: nil, pkg_deprecated: true }
+]
+
+begin
+  gem 'active_support'
+rescue Gem::LoadError
+  puts ' -> installing gem activesupport'.orange
+  Gem.install('activesupport')
+  gem 'activesupport'
+end
+require 'active_support/core_ext/object/blank'
+
+pkg_update_arr.each do |pkg|
+  next unless @device[:installed_packages].any? { |elem| elem[:name] == pkg[:pkg_name] }
+
+  puts "#{pkg[:pkg_name]} found in package fixup list".orange
+
+  # Package rename.
+  unless pkg[:pkg_rename].blank?
+    puts "#{pkg[:pkg_name]} is being renamed to #{pkg[:pkg_rename]}".orange
+    # maybe put this in a function? e.g.,
+    # rename_pkg pkg[:pkg_name] pkg[:pkg_rename]
+    old_filelist = File.join(CREW_META_PATH, "#{pkg[:pkg_name]}.filelist")
+    new_filelist = File.join(CREW_META_PATH, "#{pkg[:pkg_rename]}.filelist")
+    old_directorylist = File.join(CREW_META_PATH, "#{pkg[:pkg_name]}.directorylist")
+    new_directorylist = File.join(CREW_META_PATH, "#{pkg[:pkg_rename]}.directorylist")
+    # Handle case of new package already installed.
+    if @device[:installed_packages].any? { |elem| elem[:name] == pkg[:pkg_rename] }
+      puts "Renamed #{pkg[:pkg_rename]} is already installed. Deleting old package (#{pkg[:pkg_rename]}) information...".lightblue
+      FileUtils.rm old_filelist if File.file?(old_filelist)
+      FileUtils.rm old_directorylist if File.file?(old_directorylist)
+      @device[:installed_packages].delete_if { |elem| elem[:name] == pkg[:pkg_name] }
+      File.write "#{CREW_CONFIG_PATH}/device.json", JSON.pretty_generate(JSON.parse(@device.to_json))
+      next
+    end
+    # Handle case of package needing to be replaced.
+    if File.file?(new_filelist)
+      puts "new filelist for #{pkg[:pkg_rename]} already exists!"
+      next
+    end
+    if File.file?(new_directorylist)
+      puts "new directorylist for #{pkg[:pkg_rename]} already exists!"
+      next
+    end
+    # If new filelist or directorylist do not exist and new package is not
+    # marked as installed in device.json then rename and edit device.json .
+    puts "Renaming #{pkg[:pkg_name]} to #{pkg[:pkg_rename]}".lightblue
+    begin
+      FileUtils.cp "#{CREW_CONFIG_PATH}/device.json", "#{CREW_CONFIG_PATH}/device.json.bak"
+      FileUtils.mv old_filelist, new_filelist
+      FileUtils.mv old_directorylist, new_directorylist
+      # This following line isn't working to rename inside the data
+      # structure pulled from the json file.
+      @device[:installed_packages].map { |x| x[:name] == pkg[:pkg_name] ? pkg[:pkg_rename] : x[:name] }
```

Comment:
I think you mean this:
```ruby
@device[:installed_packages].map do |x|
  x[:name] = pkg[:pkg_rename] if x[:name] == pkg[:pkg_name]
  next x
end
```

Reply:
OMFG I added a `=`...

Reply:
Thanks for this... I was tearing my hair out over this, metaphorically speaking...